src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,74 @@
 package core.basesyntax;
 
+import java.io.*;
+
 public class WorkWithFile {
+    private static final int TRANSACTION_TYPE_INDEX = 1;
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String COMMA = ",";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeTextToFile(
+                createReport(getTextFromFile(fromFileName)),
+                toFileName);
+
+    }
+
+    private void writeTextToFile(String data, String toFileName){
```

Comment:
The access modifier for 'writeTextToFile' method should not be public since it is only used within the class. Consider changing it to private to better encapsulate the class's behavior.